### PR TITLE
fix: make sure node is drained before rebooting when rebootDelay is g…

### LINKED ISSUE #929: Kured should verify if node is still drained when delaying reboot
Hey, 

we had a problem with two controllers draining/uncordon nodes. To make the reboot more graceful, we configured `rebootDelay: 120s`. During these 2 mins the other controller uncordened the node. Kured rebooted the not drained node afterwards which is not optimal :D
So Kured should verify after the delay and right before the reboot is issued, if the node is still drained or not.

### PR DESCRIPTION
**Pull request by sebastiangaiser**
…reater than 0

closes #929

### CODE REVIEW

**Comment by jackfrancis** [comment]
I've added some concerns in the original issue #929. Let's definitely figure out how to solve for this scenario, but I'm not sure that a second drain operation after the retry wait is the right solution.

File: cmd/kured/main.go [hunk 1]
```diff
@@ -788,6 +788,18 @@ func rebootAsRequired(nodeID string, booter reboot.Reboot, sentinelCommand []str
 		if rebootDelay > 0 {
 			log.Infof("Delaying reboot for %v", rebootDelay)
 			time.Sleep(rebootDelay)
+
```
Comment (evrardjp) [inline cmd/kured/main.go:791]: I don't think it make sense to add this conditionally on reboot delay.

File: cmd/kured/main.go [hunk 2]
```diff
@@ -788,6 +788,18 @@ func rebootAsRequired(nodeID string, booter reboot.Reboot, sentinelCommand []str
...
+			log.Infof("Making sure %s is drained before reboot", node.GetName())
+			err = drain(client, node)
```
Comment (evrardjp) [inline cmd/kured/main.go:793]: I don' think we should trigger another drain. 

The state should be as _expected_ before reboot, but not triggering an extra drain. See @jackfrancis 's opinion on the issue.

**Review by evrardjp — CHANGES_REQUESTED** [review]
I don't think it's wise to add this _as is_. Let's discuss on the issue first maybe?

**Comment by evrardjp** [comment]
I like the idea, but I think we have details to iron out.

**Comment by sebastiangaiser** [comment]
Not stale...

**Comment by evrardjp** [comment]
@sebastiangaiser  maybe we could have a meeting and you explain your point of view/what's missing to progress this? I like the concept.

**Comment by sebastiangaiser** [comment]
I'm sorry but I don't have the time, currently. Feel free to hijack/supersede my PR.

**Pull request closed without merging**